### ISSUE TITLE
Can't install

### ISSUE THREAD

**Issue by lucaswxp** [thomaswinckell/ts-serialize#4]
npm install gives me:
```
├── UNMET PEER DEPENDENCY scalts@1.1.*
├── UNMET PEER DEPENDENCY scalts-array@0.7.*
├── UNMET PEER DEPENDENCY ts-json-definition@0.0.*
└── ts-serialize@1.3.0 
```

**Comment by thomaswinckell** [comment]
Yes, this is what I wanted for the `v1`. Those librairies (`scalts`, `scalts-array`, ...) should not be  dependencies of this project, because then, if you want to upgrade them in your project, you cannot because it's a dependency of a dependency. That's why  I used the peer dependencies.

There is a good answer for the question `Why use peer dependencies ?` here : https://stackoverflow.com/questions/26737819/why-use-peer-dependencies-in-npm-for-plugins#answers

So if you want to solve your issue, you need to install those dependencies (yes, it should be in the Readme, shame on me...)

TL;DR 
```
npm i --save scalts
npm i --save scalts-array
npm i --save ts-json-definition
```

By the way, I am writing a new version that will not depend on those librairies anymore. Also, there will be no more peer dependencies..

**Comment by lucaswxp** [comment]
Is there any "unserialize" method? Where I can convert JSON back to the entity? The documentation doesnt look clear.

**Comment by thomaswinckell** [comment]
You can do it using the static method `fromJsObejct` : 
```YourEntity.fromJsObject(yourJson)```

Yes, the documentation is not clear.

In V2, we will have a ```fromJson``` method.

**Comment by lucaswxp** [comment]
Sorry to bring this up again.

I still cant use it, I get:
`src/virtualServers/virtualServer.ts(3,69): error TS2307: Cannot find module 'ts-serialize'.`

But it's installed:

```
npm list | grep "ts-serialize"
├── ts-serialize@1.3.1
```

I checked node_modules, the dir is in there, but only README file is present.

**Comment by thomaswinckell** [comment]
@lucaswxp Thank you for reporting ! Sorry if you lost time with this issue.

I've just fixed the TravisCI configuration. Somehow, it doesn't work anymore without `deploy.skip_cleanup: true`.

Now it's fixed, and I've just published the version `1.3.2`. I've just checked, it's working, you can use it.

**Comment by lucaswxp** [comment]
Your are using the constructor/new to unmarshall the object?

**Comment by thomaswinckell** [comment]
@lucaswxp Yes, I need the constructor to retrieve the metadata set by the decorators.

**Comment by lucaswxp** [comment]
@thomaswinckell So you are calling "new"? My constructor has business logic that mutates object state. The object should be unmarshalled without calling new/constructor. Like this:

```
function MyClass(){}
MyClass.prototype.getName = function(){ alert(this.name); }

// unmarshall
var serializedObj = {name: "Lucas"}
serializedObj.__proto__ = MyClass.prototype
serializedObj.getName()
console.log(serializedObj instanceof MyClass)
```

**Comment by thomaswinckell** [comment]
@lucaswxp Yes, i am calling the "new". I agree, it's not a good way to unmarshall and your way seems safer. 

In fact, I did it to support the default values. When you're declaring a default value for a class property, even if it's outside the constructor, after typescript code is compiled to javascript, I assume that the default property logic ends up in the constructor. So, without calling the "new", I don't know how to retrieve the default values. 

If you have any idea to support default values without using the "new", feel free to create an issue to explain it or even do a pull request. Normally, the tests will fail if default values are not supported anymore.

**Comment by lucaswxp** [comment]
@thomaswinckell That's a problem for sure, also don't have ideas how to solve it =/
